pontoon/translate/static/js/translate.es6

```diff
@@ -0,0 +1,499 @@
+  }
+}
+
+
+/**
+ * Root Component for the entire translation editor.
+ */
+class TranslationEditor extends PontoonComponent {
+  constructor(props) {
+    super(props)
+    this.state = {
+      loaded: false,
+      sidebarOpen: true,  // Temporarily defaults to open for dev.
+      entities: [],
+      selectedIndex: 0,
+    };
+  }
+
+  componentDidMount() {
+    Entity.fetchAll(this.props.project.slug, this.props.locale.code).then((entities) => {
+      this.setState({
+        loaded: true,
+        entities: entities,
+      });
+    });
+  }
+
+  render() {
+    if (!this.state.loaded) {
+      return <LoadingScreen />;
+    }
+
+    return (
+      <div id="translation-editor">
+        <aside id="sidebar"
+               className={classNames('advanced', {open: this.state.sidebarOpen})}
+               ref="sidebar">
+          <EntityList entities={this.state.entities}
+                      project={this.props.project}
+                      selectedIndex={this.state.selectedIndex}
+                      onSelectEntity={this.selectEntity.bind(this)} />
+          <EntityDetails entity={this.selectedEntity}
+                         onSelectNextEntity={this.selectNextEntity.bind(this)}
+                         onSelectPreviousEntity={this.selectPreviousEntity.bind(this)}
+                         onSave={this.saveTranslation.bind(this)} />
+          <div id="drag" draggable="true"></div>
+        </aside>
+      </div>
+    );
+  }
+
+  get selectedEntity() {
+    return this.state.entities[this.state.selectedIndex];
+  }
+
+  selectEntity(entityKey) {
+    let index = this.state.entities.findIndex((e) => e.pk === entityKey);
+    if (index !== -1) {
+      this.setState({
+        selectedIndex: index,
+      });
+    }
+  }
+
+  saveTranslation(string) {
+    let entity = this.selectedEntity;
+
+    entity.updateTranslation(string, this.props.locale.code).then((newEntity) => {
+      console.log('save succeeded');
+      this.setState((state, props) => {
+        let index = state.entities.findIndex((e) => e.pk === entity.pk);
+        state.entities[index] = newEntity;
+        return {entities: state.entities}
+      })
+    }).fail(() => {
+      console.log('save failed');
+    });
+  }
+
+  selectNextEntity(e) {
+    e.preventDefault();
+    if (this.state.selectedIndex < this.state.entities.length - 1) {
+      this.setState({
+        selectedIndex: this.state.selectedIndex + 1,
+      });
+    }
+  }
+
+  selectPreviousEntity(e) {
+    e.preventDefault();
+    if (this.state.selectedIndex > 0) {
+      this.setState({
+        selectedIndex: this.state.selectedIndex - 1,
+      });
+    }
+  }
+
+  toggleSidebar() {
+    this.setState((state, props) => ({
+      sidebarOpen: !state.sidebarOpen
+    }));
+  }
+}
+
+
+class LoadingScreen extends PontoonComponent {
+  render() {
+    return (
+      <div id="project-load">
+        <div className="inner">
+          <div className="animation">
+            <div></div>
```

Comment:
Hey man it's what the current CSS works with and I'm not worried about greatly improving the HTML right now.